conformance/lang/expressions/multiplicative-expr/multiplication.balt

```diff
@@ -0,0 +1,3340 @@
+Labels: multiplicative-expr, int, DecimalNumber, type-cast-expr, typeof-expr, typedesc, unary-plus, unary-minus, unary-complement
+        additive-expr 
+
+public function main() {
+    int k = +4 * -3;
+    io:println(k); // @output -12
+
+    int l = -(-2 * 5);
+    io:println(l); // @output 10
+
+    int m = ~(2 * 5);
+    io:println(m); // @output -11
+
+    int n = ~2 * 5;
+    io:println(n); // @output -15
+}
+
+Test-Case: output
+Description: Test 1 for multiplicative expression in integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber
+
+public function main() {
+    int o = 2 * 404 % 10 / 4;
+    io:println(o); // @output 2
+}
+
+Test-Case: output
+Description: Test 2 for multiplicative expression in integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber
+
+public function main() {
+    int o = 2 * (404 % 10 / 4);
+    io:println(o); // @output 2
+}
+
+Test-Case: output
+Description: Test additive expressions as integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber, additive-expr 
+
+public function main() {
+    int p = (1 + 2) * (3 + 4);
+    io:println(p); // @output 21
+}
+
+Test-Case: output
+Description: Test shift expressions as integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber, shift-expr 
+
+public function main() {
+    int p = (1 << 2) * (100 >> 4);
+    io:println(p); // @output 24
+}
+
+Test-Case: output
+Description: Test binary bitwise expressions as integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber, binary-bitwise-expr 
+
+public function main() {
+    int p = (2 & 6) * (2 ^ 6);
+    io:println(p); // @output 8
+}
+
+Test-Case: output
+Description: Test conditional expressions as integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber, conditional-expr, union-type, error, optional-type
+
+public function main() {
+    int|error a = 12;
+    int? b = ();
+    int c = (a is error ? 2 : a) * (b ?: 13);
+    io:println(c); // @output 156
+}
+
+Test-Case: output
+Description: Test checking expressions as integer multiplication operands.
+Labels: multiplicative-expr, int, DecimalNumber, checking-expr, union-type, error
+
+public function main() returns error? {
+    int|error a = 12;
+    int c = check a * checkpanic a;
+    io:println(c); // @output 144
+}
+
+Test-Case: output
+Description: Test integer multiplication with operands of different int subtypes.
+Labels: multiplicative-expr, int, DecimalNumber, byte, int:Signed8, int:Signed16,
+        int:Signed32, int:Unsigned8, int:Unsigned16, int:Unsigned32, module-type-defn, union-type
+
+type Ints -9|9;
+
+public function main() returns error? {
+    int a = -1;
+    byte b = 2;
+    int:Signed8 c = -3;
+    int:Signed16 d = 4;
+    int:Signed32 e = -5;
+    int:Unsigned8 f = 6;
+    int:Unsigned16 g = 7;
+    int:Unsigned32 h = 8;
+    Ints i = -9;
+
```

Comment:
Shall we also add some tests like
```ballerina
type Ints int:Signed16|int:Signed32;
public function main() {
    int:Signed16|int:Signed32 a = 12;
    Ints b = 24;
    int c = a * b;
}
```

Reply:
Thanks, I had missed this. Created an issue and added tests.